File: programming_hw_4s/tasks_with_eolymp_tags/t_8_8_(eolimp_1457).py
def analyze(arr, arr_size, p):
    for i in range(1, arr_size):
        j = i
        tmp = arr[i]

        while (j > 0) and (arr[j - 1] > arr[j]):
            if p < (arr[j - 1] + tmp): return "No"
            arr[j] = arr[j - 1]
            j -= 1

        arr[j] = tmp

    return "Yes"


if __name__ == "__main__":
    n, m = map(int, input().split())
    cars = list(map(int, input().split()))

    print(analyze(cars, n, m), end='')

File: programming_hw_4s/tasks_with_eolymp_tags/t_8_8_(eolimp_1457).cpp
#include <iostream>
#include <vector>

using namespace std;

string analyze(vector<int> &arr, int arr_size, int p){
	int i, j;
	int tmp;

	for (i = 1; i < arr_size; i++){
		j = i;
		tmp = arr[i];

		while (j > 0 && arr[j - 1] > arr[j]){
			if (p < arr[j - 1] + tmp) return "No";
			arr[j] = arr[j - 1];
			j--;
		}

		arr[j] = tmp;
	}

	return "Yes";
}

int main(){
	int n, m;
	vector<int> cars;
	cin >> n >> m;
	cars.resize(n);
	for (int c = 0; c < n; c++) cin >> cars[c];

	cout << analyze(cars, n, m);
	return 0;
}